### DISCUSSION TITLE
ESP32 resetet sich beim Zugriff via WebGUI

### DISCUSSION THREAD

**Discussion by Noxytech** [beabel/radiowecker#15] [Fragen & Antworten / Q&A]
Hallo,
ich habe ein merkwürdiged Phänomen mit dem ESP32 beim Zugriff via WebGUI. 
Sobalds ich versuche via Browser (Firefox, Chrome, Edge) auf das Web Frontend zu zugreifen, steht im Display "Connection lost" und der ESP32 startet neu.
Das gleiche passiert auch mit einem Nagelneuen ESP32. Aufgefallen ist mir das ganze mit Version 2.1.0 und auch mit der davor (2.0.4).
Ich benutze Windows 10 Pro, Arduino IDE 1.8.19, WLAN-AP ist eine Fritzbox 7590 und der Abstand zur Box ist ca 3m Luftlinie und sichtbar. Probiert habe ich den ESP32 (ich meine den ESP32 im AZ-Delivery Radiowecker Bausatz) mit einem 9V/1A Netzteil als auch über USB (5V/1A) mit Strom zu speisen, bei beiden ESP32 das gleiche Problem. Spiele ich aber die drittletzte Version rauf, funktioniert der Zugriff auf das WebGUI.
Weiß jemand was das sein könnte? 
Für jeden Tipp bin ich dankbar.

Gruß
Harald

**Comment by beabel** [comment]
Also das hört sich komisch an.
Ich habe bei mir nur festgestellt, das es mehrere Versionen des ESP Dev Moduls gibt.
Mit dem ESP32-S habe ich sehr schlechten WLAN Empfang, wobei ich mit dem ESP32-WROOM-32 damit keine Probleme habe.
Der unterschied dabei ist auch der ESP32-S verbindet sich automatisch beim Übertragen per USB, beim WROOM muss ich immer eine Taste drücken.
![IMG20231111082229](https://github.com/beabel/radiowecker/assets/12296286/6741efc7-1e41-4db9-8c19-f4334b05fc18)

**Comment by Noxytech** [comment]
Ich vergaß die Version meiner ESP32 zu erwähnen:
AZ-Delivery
ESP32-WROOM-32
steht auf dem Schutzblech
und  auf der Unterseite der Platine steht
ESP32_Devkitc_V4

Einen ESP32-S (Platinenunterseite: NODEMCU-32 V1.3)
von AZ-Delivery habe ich auch, der hat eine augedruckte Antenne
und einen externen Antennenanschluss.
Leider konnte ich ihn nicht testen weil das Pinlayout zu schmal ist.
Ich müsste einen Adaptersockel bauen um ihn in den Radiowecker einstecken zu können.

**Reply by Noxytech** [reply]
Nach einigen Tests bin ich zum Schluss gekommen, das seit Einbindung der ArduinoJson.h ich keinen Zugriff mehr auf die WebGUI habe. Es kommt, wie schon gesagt, auf dem Display die Meldung "Connection lost" und der Radiowecker startet nach einer Weile neu. Die zur Zeit aktuelle ArduinoJson von Benoit Blanchon Vers. 6.21.3 ist auch installiert. An der WLAN Performance (Signal -56, laut Display Anzeige) kann es nicht liegen, den ein Code upload aus der Arduino IDE via OTA läuft problemlos mit beiden ESP32's durch. Es sieht für mich so aus als würde etwas im Webserver Code, bzw. dem Anzeige Code der WebGUI passieren, das den Timeout im Browser erzeugt, und die WLAN Verbindung trennt oder resetet. Leider komme ich an diesem Punkt nicht weiter, und muss wahrscheinlich auf die neuen WebGUI Funktionen verzichten. Schade. Aber trotzdem Danke für die bisherige Entwicklung und anpassung des Radioweckers.

**Reply by Noxytech** [reply]
Ich habe mal meinen Radiowecker mit dem Online Flash Tool aktualisiert, wie in [diesem Thread](https://github.com/beabel/radiowecker/discussions/21) beschrieben. und siehe da, ich kann tatsächlich direkt auf die WebGUI zugreifen ohne das der Radiowecker sich resetet. Es läuft alles soweit gut und läßt sich auch normal bedienen über die WebGUI. Hm, was könnte dann bei mir anders sein? Aber leider kann ich ja durch das Online flashen die Displayfarbe nicht ändern und auch nicht das Update Passwort. Immerhin schon mal ein Erfolg.
